### PR TITLE
Update HubConnection url in Java client

### PR DESCRIPTION
**Pull request by mikaelm12**
Part of issue: https://github.com/aspnet/AspNetCore/issues/5342

### CODE REVIEW

File: src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java [hunk 1]
```diff
@@ -290,6 +290,27 @@ public HubConnectionState getConnectionState() {
...
+     * Sets a new url for the HubConnection.
+     * @param url The url to connect to.
+     */
+    public void setBaseUrl(String url) {
+        if (url == null || url.isEmpty()) {
+            throw new IllegalArgumentException("The HubConnection url must be a valid url.");
+        }
+
+        if (hubConnectionState != HubConnectionState.DISCONNECTED) {
+            throw new IllegalStateException("The HubConnection must be in the disconnected state to change the url.");
```
Comment (anurse) [inline src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java]: Are we guaranteed that the connection will be in the `DISCONNECTED` state during the onClosed callback? I'd like to see a test confirming that works (changing the URL in the `onClosed` callback, and doing so **without** waiting for `.stop` to complete, to make sure the asynchrony is working right).

**Comment by mikaelm12** [comment]
> Are we guaranteed that the connection will be in the DISCONNECTED state during the onClosed callback? I'd like to see a test confirming that works

Added that test and ran in in a loop locally 10 thousand times.

**Comment by mikaelm12** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=219084

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2200
https://github.com/aspnet/AspNetCore-Internal/issues/2656

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**